File: java/amp-examples/stream-subscriber/src/main/java/amp/examples/streaming/subscriber/StreamingCollectionSubscriber.java
package amp.examples.streaming.subscriber;

import amp.eventing.streaming.DefaultStreamingBus;
import amp.eventing.streaming.IStandardStreamingEventBus;
import amp.examples.streaming.common.ModernMajorGeneralMessage;
import cmf.eventing.patterns.streaming.IStreamingCollectionHandler;
import org.springframework.context.support.ClassPathXmlApplicationContext;

public class StreamingCollectionSubscriber {
    public static void main(String[] args) throws Exception {
        ClassPathXmlApplicationContext injector = new ClassPathXmlApplicationContext("classpath:META-INF/spring/eventBusContext.xml");

        IStandardStreamingEventBus streamingEventBus = injector.getBean("eventBus", DefaultStreamingBus.class);

        IStreamingCollectionHandler<ModernMajorGeneralMessage> handler = new CollectionHandler();
        streamingEventBus.subscribeToCollection(handler);

        System.in.read();
        System.exit(0);
    }
}

File: dotnet/examples/amp.examples.streaming.subscriber/StreamingCollectionSubscriber.cs
using amp.eventing.streaming;
using amp.examples.streaming.common;
using cmf.eventing.patterns.streaming;
using Spring.Context;
using Spring.Context.Support;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace amp.examples.streaming.subscriber
{
    public class StreamingCollectionSubscriber
    {
        public static void Main(string[] args)
        {
            IApplicationContext springContainer = ContextRegistry.GetContext();

            IStandardStreamingEventBus streamingEventBus = springContainer.GetObject(
                typeof(IStandardStreamingEventBus).FullName)
                as IStandardStreamingEventBus;

            IStreamingCollectionHandler<ModernMajorGeneralMessage> handler = new CollectionHandler();
            streamingEventBus.SubscribeToCollection<ModernMajorGeneralMessage>(handler);


            Console.WriteLine("If all events have been received, hit the Enter Key to exit the program.");
            Console.Read();
            Environment.Exit(0);
        }
    }
}
